### ISSUE TITLE
Downloading podcast episode appears to go into a loop

### ISSUE THREAD

**Issue by EricBrunel** [stonega/tsacdop#26]
I have an issue while trying to download an episode of this podcast:
https://anchor.fm/s/1bf05abc/podcast/rss
Trying to dowload episode 9 for example ("DEVIN TOWNSEND PODCAST #9 - The New Black") seems to work fine, goes up to 100%, and then just starts over, back at 1%. I never managed to download this episode successfully. I'm getting no error message, nothing, it just goes into a loop.
Not sure if it's an issue with Tsacdop or the podcast itself. I did manage to download the file by using a basic RSS feed aggregator though.
Any idea about what could happen?
Thanks.
 - Eric

**Comment by stonega** [comment]
Thanks for the report.
I will have a check about it!

**Comment by stonega** [comment]
Can you try again once you updated to v0.5.0? I can download the episode on my device.

**Comment by EricBrunel** [comment]
I did update and tried again, but no luck: downloads are still failing. The weird thing is that they don't seem to even reach 100%: before the end, I get a 'cancelled' notification, and the download starts over again. Is there any log I can check or send you to help figure out what's happening here?

**Comment by stonega** [comment]
Do do have the same issue with other podcasts? I have no clue what happened right now. 
If it take too long time, it might be timeout issue.

**Comment by EricBrunel** [comment]
I don't have issues with other podcasts, but this one is the one for which the episodes are the biggest, almost 200MB for one episode. All the others are way below that, like, hardly more than 100MB for an episode at most. And my internet connection is not good, usually around 2.5Mb/s. So a time-out looks like a possible reason indeed. Any way I have to check that?
Thanks!

**Comment by EricBrunel** [comment]
Today, I just had the issue with another podcast, for which I was trying to dowload an episode while my bandwidth was already quite busy. And the same thing happened: the download kept getting cancelled and started over again. I stopped it, then retried when my bandwidth was free, and it worked without problem. So it does feel like some sort of time-out somewhere...
Again, if there's any way I can help, please tell me, I'll be happy to.

**Comment by stonega** [comment]
@EricBrunel  Hi, this is almost caused by the connection timeout setting, I will change it to more longer in next update.

**Comment by EricBrunel** [comment]
Thanks for looking into it! I'll wait for the next version and tell you how it goes.

**Comment by stonega** [comment]
@EricBrunel Please let me know if you still have the same issue after updated to 0.5.1. Thanks!

**Comment by EricBrunel** [comment]
Nope, no luck. :-( I tried again with the episode I mentionned in the original post above, and it's still doing the same thing: it seems to go to 100%, but then starts over. I did it twice and paid attention to what was happening the second time though, and it seems the last segment for the download (between 89% and 100%) was shorter than the others, so not sure it actually reached 100%. I also noticed that it seemed to "jump" to 100% almost exactly 10 minutes after the beginning of the download. If it's the time-out you're using, it's not enough: my bandwidth is usually around 2.5 Mb/s, so downloading an episode of 196 MB would last 196 * 8 / 2.5 = 627.2 seconds, which is more than 10 minutes...
